internal/colorize/cropped.go

```diff
@@ -36,8 +36,33 @@ func GetCroppedText(text string, maxLen int, includeLineEnds bool) CroppedLines
 
 		// Ensure the next position is not within a color tag and check conditions that would end this entry
 		if isLineEnd || (!inRange(pos+1, colorCodes) && (entry.Length == maxLen || pos == len(text)-1)) {
+			wrapped := ""
+			wrappedLength := 0
+			nextCharIsSpace := pos+1 < len(text) && isSpace(text[pos+1])
+			if !isLineEnd && entry.Length == maxLen && !nextCharIsSpace {
+				// Put the current word on the next line, if possible.
+				// Find the start of the current word and its printed length, taking color ranges and
+				// multi-byte characters into account.
+				i := len(entry.Line) - 1
+				for ; i > 0; i-- {
+					if isSpace(entry.Line[i]) {
+						break
+					}
+					if !inRange(pos-(len(entry.Line)-i), colorCodes) && !isUTF8TrailingByte(entry.Line[i]) {
```

Comment:
nit: a comment here _might_ be helpful though it's fairly evident now that I've read it a few times. I'll leave it up to you.